### PR TITLE
Docs: Rename Block Hooks handbook page to Block Filters

### PR DESCRIPTION
**Pull request by ockham**
## What?

Rename the handbook page currently entitled "Block Hooks" to "Block Filters".

## Why?

To distinguish from the new ["Block Hooks" feature](https://core.trac.wordpress.org/ticket/59313), which is conceptually very different.

## How?

By renaming the handbook page from "Block Hooks" to "Block Filters".

AFAICT, the handbook page makes no direct reference to hooks, other than the title; furthermore, the relevant client-side `@wordpress/` package happens to be called `@wordpress/hooks`. Reading the page, it’s all about filters though (both on the client and server sides), whereas "hooks" traditionally denotes the union of filters and actions — the latter of which are pretty much absent from the page.

Finally, note that the filename has already been `block-filters.md` before.

Thus, changing the title to "Block Filters" seems warranted.

Flagged by @juanmaguitar in Slack.

### CODE REVIEW

**Comment by juanmaguitar** [comment]
LGTM! 👍 
We would also need to change the reference to this page from the landing page of [Reference Guides](https://developer.wordpress.org/block-editor/reference-guides/) though.

**Comment by ockham** [comment]
> LGTM! 👍 We would also need to change the reference to this page from the landing page of [Reference Guides](https://developer.wordpress.org/block-editor/reference-guides/) though.

Ah, good point! Done in 9b1bcdff5c.

File: docs/reference-guides/README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 ## [Hooks Reference](/docs/reference-guides/filters/README.md)
```
Comment (ockham) [inline docs/reference-guides/README.md:19]: Should we also rename this to "Hooks and Filters Reference" now? Ripple effects... 😬 

 ```suggestion
## [Hooks and Filters Reference](/docs/reference-guides/filters/README.md)
```

Comment (juanmaguitar) [reply docs/reference-guides/README.md:19]: In theory, the concept "Hooks" would already include "Filters" (and "Actions") so I think the current two changes are enough for now for the purpose of this PR. 
We can think later of a better way to name these Reference API pages

**Review by juanmaguitar — APPROVED** [review]

**Comment by ockham** [comment]
Merged.

~@juanmaguitar @ndiego Does this need any extra work to replace the current docs over at https://developer.wordpress.org/block-editor/reference-guides/filters/block-filters/?~ Disregard, looks like it's already updated 😅

**Comment by ndiego** [comment]
Yeah it updates every 15 minutes.

**Comment by gziolo** [comment]
I led the effort to consolidate documentation pages explaining the usage of WordPress hooks with the editor at https://github.com/WordPress/gutenberg/pull/42356. It doesn't always cover filters like it was called previously, so we decided to unify the vocabulary to follow WordPress core:

https://developer.wordpress.org/plugins/hooks/

It's also grouped under the "hooks" in the code reference:

https://developer.wordpress.org/reference/hooks/

The biggest difference between Gutenberg and WordPress core is that for core the documentation is generated from the source code, it's searchable, and based on individual use cases. On the contrary, only a subset of WordPress hooks available are documented in the block editor because it has to be done manually. If we were to follow WP core, then the issue wouldn't exist as we wouldn't have to divide filters and actions based on their role arbitrarily. 

In conclusion, the phrase Block Hooks is confusing here, and Block Filters is a good short-term solution, but it's the issue that we have to manually maintain code references for WordPress hooks used with the editor.
